--- tazbox.orig
+++ tazbox
@@ -37,6 +37,7 @@
   out-dl     $(gettext "Pipe Wget output into a GTK window")
   locale     $(gettext "Configure system language (root)")
   keymap     $(gettext "Configure system keymap (root)")
+  tz         $(gettext "Configure system timezone (root)")
 
 EOT
 }
@@ -154,6 +155,31 @@
 	[ "$keymap" ] && tazkeymap init $keymap
 }
 
+# TZ functions
+tz_main() {
+    text=$(gettext "TimeZone Configuration")
+    zi="/usr/share/zoneinfo/"
+    for t in $(find $zi -type f | sed s,$zi,,g)
+    do
+        echo -e "$t"
+    done | \
+    yad --list --title="SliTaz TZ" --window-icon=$icon \
+        --width=500 --height=380 --separator="" \
+        --center --sticky --on-top \
+        --image=preferences-desktop-locale --image-on-top \
+        --text="<b>$text</b>" --print-column=1 \
+        --column $(gettext "Name")
+}
+
+tz() {
+    timezone=$(tz_main)
+    case $? in
+        1) exit 0 ;;
+        *) continue ;;
+    esac
+    [ "$timezone" ] && echo $timezone > /etc/TZ
+}
+
 #
 # Commands
 #
@@ -213,6 +239,8 @@
 		locale ;;
 	keymap)
 		keymap ;;
+    tz)
+        tz ;;
 	boot)
 		# This command is used at first boot to configure system.
 		Xorg -br -quiet -nolisten tcp :1 &
